velox/type/DecimalUtil.h

```diff
@@ -147,6 +148,45 @@ class DecimalUtil {
     return static_cast<TOutput>(rescaledValue);
   }
 
+  /// Rescale a double value to decimal value.
+  ///
+  /// Use `folly::convertTo` to convert a double value to int128_t or int64_t,
+  /// it will throw an exception like 'loss of precision' when the `toValue`
+  /// overflows the limits of int128_t or int64_t. The exception will be caught
+  /// here to judge whether there is a overflow.
+  ///
+  /// Normally, return the rescaled value. Otherwise, if the `toValue` overflows
+  /// the TOutput's limits or the `toValue` exceeds the precision's limits, it
+  /// will throw an exception.
+  template <typename TOutput>
+  inline static std::optional<TOutput> rescaleDouble(
```

Comment:
Why is the output an optional? When do you return unset optional?

It would be nice to extract this utility into a separate PR and add tests. CC: @majetideepak